components/entitlement/org.wso2.carbon.identity.entitlement.ui/src/main/resources/web/entitlement/authorization-index.jsp

```diff
@@ -239,23 +240,26 @@
             %>
             <tr>
                 <td style="width:100px;">
-                    <a class="icon-link" onclick="updownthis(this,'up',<%=request.getParameter("pageNumber")%>,<%=numberOfPages%>)" style="background-image:url(../admin/images/up-arrow.gif)"></a>
-                    <a class="icon-link" onclick="updownthis(this,'down',<%=request.getParameter("pageNumber")%>,<%=numberOfPages%>)" style="background-image:url(../admin/images/down-arrow.gif)"></a>
+                    <a class="icon-link" onclick="updownthis(this,'up',<%=Encode.forHtmlContent(request.getParameter("pageNumber"))%>,
+                            <%=numberOfPages%>)" style="background-image:url(../admin/images/up-arrow.gif)"></a>
+                    <a class="icon-link" onclick="updownthis(this,'down',<%=Encode.forHtmlContent(request.getParameter("pageNumber"))%>,
+                            <%=numberOfPages%>)" style="background-image:url(../admin/images/down-arrow.gif)"></a>
                     <input type="hidden" value="<%=policies[i].getPolicyId()%>"/>                    
                 </td>
 
                 <td width="10px" style="text-align:center; !important">
                     <input type="checkbox" name="policies"
-                           value="<%=policies[i].getPolicyId()%>"
+                           value="<%=Encode.forHtmlAttribute(policies[i].getPolicyId())%>"
                            onclick="resetVars()" class="chkBox" <% if(!delete){%>disabled="disabled"<% } %>/>
                 </td>
 
                 <td>
-                    <a <% if(edit) { %>href="policy-view.jsp?policyid=<%=policies[i].getPolicyId()%>" <% } %>><%=policies[i].getPolicyId()%></a>
+                    <a <% if(edit) { %>href="policy-view.jsp?policyid=<%=Encode.forUriComponent(policies[i].getPolicyId())%>" <% } %>>
+                        <%=Encode.forHtmlContent(policies[i].getPolicyId())%></a>
                 </td>
 
                 <td width="40%">
-                    <a title="<fmt:message key='edit.policy'/>" onclick="edit('<%=policies[i].getPolicyId()%>');return false;"
+                    <a title="<fmt:message key='edit.policy'/>" onclick="edit('<%=Encode.forJavaScriptAttribute(policies[i].getPolicyId())%>');return false;"
```

Comment:
uri encode and then apply javascript attribute encode, since the passed parameter is used in a query string